include/deal.II/base/exceptions.h

```diff
@@ -1130,6 +1130,25 @@ namespace StandardExceptions
 #define AssertIsFinite(number) Assert(dealii::numbers::is_finite(number), \
                                       dealii::ExcNumberNotFinite(std::complex<double>(number)))
 
+#ifdef DEAL_II_WITH_CUDA
+/**
+ * An assertion that checks that the error code produced by calling a CUDA
+ * routine is equal to cudaSuccess.
+ *
+ * @ingroup Exceptions
+ * @author Bruno Turcksin, 2016
+ */
+#define CudaAssert(error_code)                                                \
+  {                                                                           \
+    if (cudaSuccess != error_code)                                            \
+      {                                                                         \
+        fprintf(stderr,"Error in %s (%d): %s\n",__FILE__,                       \
+                __LINE__,cudaGetErrorString(error_code));                       \
+        exit(1);                                                                \
```

Comment:
Then that would mean that we do not properly pass these flags to the compiler's command line. Other bad things are likely going to happen in that case downstream, so it would be worth investigating what exactly it is that is happening.